relay-metrics/src/aggregation.rs

```diff
@@ -811,15 +820,22 @@ impl AggregatorConfig {
     ///
     /// Recent buckets are flushed after a grace period of `initial_delay`. Backdated buckets, that
     /// is, buckets that lie in the past, are flushed after the shorter `debounce_delay`.
-    fn get_flush_time(&self, bucket_timestamp: UnixTimestamp) -> Instant {
+    fn get_flush_time(&self, bucket_timestamp: UnixTimestamp, project_key: ProjectKey) -> Instant {
         let now = Instant::now();
 
         if let MonotonicResult::Instant(instant) = bucket_timestamp.to_instant() {
             let bucket_end = instant + self.bucket_interval();
             let initial_flush = bucket_end + self.initial_delay();
+            // If the initial flush is still pending, use that.
             if initial_flush > now {
-                // If the initial flush is still pending, use that.
-                return initial_flush;
+                // Shift deterministically within one bucket interval based on the project key. This
+                // distributes buckets over time while also flushing all buckets of the same project
+                // key together.
+                let mut hasher = FnvHasher::default();
+                hasher.write(project_key.as_str().as_bytes());
+                let shift_millis = u64::from(hasher.finish()) % (self.bucket_interval * 1000);
+
+                return initial_flush + Duration::from_millis(shift_millis);
```

Comment:
I wonder what percentage of our buckets is flushed with `debounce_delay` rather than `initial_delay`. Might be worth adding a metric for that in a followup PR. In case it's high, do we need jitter on the debounce delay as well?